foundations/foundation-metrics/src/main/java/io/servicecomb/foundation/metrics/MetricsServoRegistry.java

```diff
@@ -0,0 +1,463 @@
+   */
+  protected final Func0<String> getCpuAndMemory = new Func0<String>() {
+    @Override
+    public String call() {
+      Map<String, String> memoryMap = new HashMap<>();
+      OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
+      double cpu = osMxBean.getSystemLoadAverage();
+      memoryMap.put("CPU System Load", String.valueOf(cpu));
+
+      ThreadMXBean threadmxBean = ManagementFactory.getThreadMXBean();
+      int threadCount = threadmxBean.getThreadCount();
+      memoryMap.put("CPU Current Running Threads", String.valueOf(threadCount));
+
+      MemoryMXBean memBean = ManagementFactory.getMemoryMXBean();
+      MemoryUsage memHeapUsage = memBean.getHeapMemoryUsage();
+      MemoryUsage nonHeapUsage = memBean.getNonHeapMemoryUsage();
+      memoryMap.put("heapInit", String.valueOf(memHeapUsage.getInit()));
+      memoryMap.put("heapMax", String.valueOf(memHeapUsage.getMax()));
+      memoryMap.put("heapCommit", String.valueOf(memHeapUsage.getCommitted()));
+      memoryMap.put("heapUsed", String.valueOf(memHeapUsage.getUsed()));
+      memoryMap.put("nonHeapInit", String.valueOf(nonHeapUsage.getInit()));
+      memoryMap.put("nonHeapMax", String.valueOf(nonHeapUsage.getMax()));
+      memoryMap.put("nonHeapCommit", String.valueOf(nonHeapUsage.getCommitted()));
+      memoryMap.put("nonHeapUsed", String.valueOf(nonHeapUsage.getUsed()));
+      return memoryMap.toString();
+    }
+  };
+
+  /**
+   * Get TPS and latency for operational and instance level from hystrix.
+   */
+  protected final Func0<String> getTpsAndLatency = new Func0<String>() {
+    @Override
+    public String call() {
+      Map<String, String> tpsAndLatencyMap = new HashMap<>();
+      Collection<HystrixCommandMetrics> instances = HystrixCommandMetrics.getInstances();
+
+      long insTotalTps = 0;
+      long insTotalLatency = 0;
+
+      for (HystrixCommandMetrics instance : instances) {
+        long successCount = instance.getRollingCount(HystrixEventType.SUCCESS);
+        long failureCount = instance.getRollingCount(HystrixEventType.FAILURE);
+        int operLatency = instance.getExecutionTimeMean();
+        long totalCallCount = successCount + failureCount;
+
+        int windowTime = instance.getProperties().metricsRollingStatisticalWindowInMilliseconds().get() / 1000;
+        double qpsVal = (double) (totalCallCount) / windowTime;
+        BigDecimal bigDecimal = new BigDecimal(qpsVal);
+        BigDecimal bigDecimalVal = bigDecimal.setScale(1, RoundingMode.HALF_DOWN);
+        Double tpsOper = bigDecimalVal.doubleValue();
+        String operPath = MetricsDataMonitor.getMetricsGlobalRef()
+            .getHystrixHelpMap(instance.getCommandKey().name());
+        tpsAndLatencyMap.put("TPS-" + operPath, String.valueOf(tpsOper));
+        tpsAndLatencyMap.put("Latency-" + operPath, String.valueOf(operLatency));
+        insTotalTps += tpsOper;
+        insTotalLatency += operLatency;

```

Comment:
I think we can not direct plus all latency together，may plus (operLatency * rolling count percentage ) as result

Reply:
fixed